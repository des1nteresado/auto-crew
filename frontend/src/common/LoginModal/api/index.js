import api from '../../../api/config';

export const signIn = (body) => api.post('/auth/sign_in', body);
