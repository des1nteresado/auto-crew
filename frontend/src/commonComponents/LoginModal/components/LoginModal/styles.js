const styles = (theme) => ({
  closeButton: {
    minWidth: 20,
    maxWidth: 20,
    height: 40,
    position: 'absolute',
    top: 11,
    right: 15,
    padding: 0,
  },

  commomDialog: { borderRadius: 10 },
  dialogWrapper: {
    maxWidth: 324,
    textAlign: 'center',
    margin: 'auto',
    maginBotom: 53,
    [theme.breakpoints.down(420)]: {
      margin: -10,
    },
  },
  resetPassword: {
    fontSize: 12,
    marginLeft: 'auto',
  },
  passwordBox: {
    textAlign: 'right',
  },
  signUpButton: {
    fontSize: 12,
  },
  topActions: {
    marginBottom: 10,
    [theme.breakpoints.down('xs')]: {
      '& .MuiTypography-h2': {
        fontSize: 19,
      },
      '& .MuiTypography-h3': {
        fontSize: 16,
      },
    },
  },
});

export default styles;
