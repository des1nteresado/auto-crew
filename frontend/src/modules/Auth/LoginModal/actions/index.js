import { createAction } from 'redux-actions';

export const LOGIN = createAction('LOGIN_REQUEST');
export const LOGIN_SUCCESS = createAction('LOGIN_SUCCESS');
export const LOGIN_FAIL = createAction('LOGIN_FAIL');

export const CLEAR_ERRORS = createAction('CLEAR_ERRORS');
