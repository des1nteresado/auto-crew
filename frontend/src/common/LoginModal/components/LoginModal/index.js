import React from 'react';
import PropTypes from 'prop-types';
import { Grid, Typography } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import CloseIcon from '@material-ui/icons/Close';

import CommonDialog from 'common/components/CommonDialog';
import CommonButton from 'common/components/CommonButton';
import CommonInput from 'common/components/CommonInput';
import { INVALID_EMAIL, INVALID_BASE_INPUT } from 'constants/errorMessages';

import styles from './styles';

const LoginModal = (props) => {
  const {
    formData: { email, password },
    onChange,
    onSubmit,
    isEmailValid,
    isPasswordValid,
    classes,
    isLoginModalShowed,
    handleShowLoginModal,
    handleShowResetPasswordModal,
    handleShowRegisterModal,
  } = props;

  return (
    <CommonDialog
      isOpen={isLoginModalShowed}
      style={classes.commomDialog}
      onClose={() => handleShowLoginModal(false)}
    >
      <form onSubmit={onSubmit}>
        <CommonButton
          className={classes.closeButton}
          startIcon={<CloseIcon />}
          label=""
          variant="text"
          size="large"
          onClick={() => handleShowLoginModal(false)}
        />
        <Grid container spacing={4} className={classes.dialogWrapper}>
          <Grid container item spacing={4} xs={12} className={classes.topActions}>
            <Grid item xs={12}>
              <Typography variant="h2" color="primary">
                Welcome back!
              </Typography>
            </Grid>
            <Grid item xs={12}>
              <Typography variant="h3" color="primary">
                Please sign in.
              </Typography>
            </Grid>
          </Grid>
          <Grid item xs={12}>
            <CommonInput
              value={email}
              onChange={onChange}
              name="email"
              label="Email"
              autoFocus
              error={email != '' && !isEmailValid}
              helperText={INVALID_EMAIL}
            />
          </Grid>
          <Grid item xs={12} className={classes.passwordBox}>
            <CommonInput
              type="password"
              value={password}
              onChange={onChange}
              name="password"
              label="Password"
              error={password != '' && !isPasswordValid}
              helperText={INVALID_BASE_INPUT}
            />
            <CommonButton
              label=" Forgot your password?"
              variant="text"
              className={classes.resetPassword}
              onClick={handleShowResetPasswordModal}
            />
          </Grid>
          <Grid item xs={12}>
            <CommonButton
              label="Sign in"
              variant="contained"
              disabled={!(isEmailValid && isPasswordValid)}
              type="submit"
            />
            <CommonButton
              label="Create an account"
              variant="text"
              className={classes.signUpButton}
              onClick={handleShowRegisterModal}
            />
          </Grid>
        </Grid>
      </form>
    </CommonDialog>
  );
};

LoginModal.propTypes = {
  classes: PropTypes.oneOfType([PropTypes.object]).isRequired,
  formData: PropTypes.oneOfType([PropTypes.object]).isRequired,
  isEmailValid: PropTypes.bool.isRequired,
  isPasswordValid: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  isLoginModalShowed: PropTypes.bool.isRequired,
  handleShowLoginModal: PropTypes.func.isRequired,
  handleShowResetPasswordModal: PropTypes.func.isRequired,
  handleShowRegisterModal: PropTypes.func.isRequired,
};

export default React.memo(withStyles(styles)(LoginModal));
