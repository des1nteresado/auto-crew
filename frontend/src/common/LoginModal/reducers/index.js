import { handleActions } from 'redux-actions';

import * as actions from '../actions/index';
import { UPDATE_USER_PROFILE_SUCCESS } from '../../../modules/ProfilePage/components/Tabs/ProfileTab/actions';

const defaultState = {
  _id: null,
  orders: [],
  follows: [],
  email: null,
  firstName: '',
  lastName: '',
  role: '',
  isLoading: false,
  token: '',
  isAuth: false,
  errors: '',
};

const userReducer = handleActions(
  {
    [actions.LOGIN]: (state) => ({
      ...state,
      isLoading: true,
    }),
    [actions.LOGIN_SUCCESS]: (state, { payload }) => {
      if (payload && payload.response && payload.response.user) {
        const { token, user } = payload.response;
        const { _id, orders, email, firstName, lastName, role, follows } = user;

        return {
          ...state,
          isLoading: false,
          _id,
          orders,
          follows,
          email,
          firstName,
          lastName,
          role,
          token,
          isAuth: true,
          errors: null,
        };
      }

      return defaultState;
    },
    [actions.LOGIN_FAIL]: (state, { payload }) => ({
      ...state,
      isLoading: false,
      isAuth: false,
      errors: payload && payload.response,
    }),
    [actions.CLEAR_ERRORS]: (state) => ({
      ...state,
      errors: '',
    }),
    [UPDATE_USER_PROFILE_SUCCESS]: (state, { payload = {} }) => {
      const { firstName, lastName, email, follows } = payload.response;

      return {
        ...state,
        firstName,
        lastName,
        email,
        follows,
      };
    },
  },
  defaultState
);

export default userReducer;
