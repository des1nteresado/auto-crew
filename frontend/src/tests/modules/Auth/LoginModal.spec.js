import React from 'react';
import { shallow } from 'enzyme';

import LoginModal from 'modules/Auth/LoginModal/components/LoginModal';

const defaultProps = {
  formData: { email: 'testemail@mail.ru', password: 'password' },
  onChange: jest.fn(),
  onSubmit: jest.fn(),
  isEmailValid: true,
  isPasswordValid: true,
  isLoginModalShowed: true,
  handleShowLoginModal: jest.fn(),
  handleShowResetPasswordModal: jest.fn(),
  handleShowRegisterModal: jest.fn(),
};

const setUp = (props) => shallow(<LoginModal {...defaultProps} {...props} />).dive();

describe('LoginModal', () => {
  let component;

  beforeEach(() => {
    component = setUp();
  });

  afterEach(() => {
    defaultProps.handleShowLoginModal.mockRestore();
  });

  it('renders LoginModal component correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('calls submit callback correctly', () => {
    const form = component.find('form');

    form.simulate('submit');

    expect(defaultProps.onSubmit).toHaveBeenCalled();
  });

  it('calls onClose CommonDialog callback correctly', () => {
    const commonDialog = component.find('Memo(WithStyles(CommonDialog))');

    expect(commonDialog.props().isOpen).toBe(defaultProps.isLoginModalShowed);

    commonDialog.props().onClose();

    expect(defaultProps.handleShowLoginModal).toHaveBeenCalledWith(false);
  });

  it('calls onClick CloseButton callback correctly', () => {
    const closeButton = component.find('Memo(CommonButton)').first();

    closeButton.simulate('click');

    expect(defaultProps.handleShowLoginModal).toHaveBeenCalledWith(false);
  });
});
