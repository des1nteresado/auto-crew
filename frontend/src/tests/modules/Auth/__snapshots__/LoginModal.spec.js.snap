// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginModal renders LoginModal component correctly 1`] = `
<Memo(WithStyles(CommonDialog))
  isOpen={true}
  onClose={[Function]}
  style="LoginModal-commomDialog-2"
>
  <form
    onSubmit={[MockFunction]}
  >
    <Memo(CommonButton)
      className="LoginModal-closeButton-1"
      label=""
      onClick={[Function]}
      size="large"
      startIcon={<UNDEFINED />}
      variant="text"
    />
    <WithStyles(ForwardRef(Grid))
      className="LoginModal-dialogWrapper-3"
      container={true}
      spacing={4}
    >
      <WithStyles(ForwardRef(Grid))
        className="LoginModal-topActions-7"
        container={true}
        item={true}
        spacing={4}
        xs={12}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={12}
        >
          <WithStyles(ForwardRef(Typography))
            color="primary"
            variant="h2"
          >
            Welcome back!
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={12}
        >
          <WithStyles(ForwardRef(Typography))
            color="primary"
            variant="h3"
          >
            Please sign in.
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <Memo(WithStyles(CommonInput))
          autoFocus={true}
          error={false}
          helperText="Invalid email address."
          label="Email"
          name="email"
          onChange={[MockFunction]}
          value="testemail@mail.ru"
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        className="LoginModal-passwordBox-5"
        item={true}
        xs={12}
      >
        <Memo(WithStyles(CommonInput))
          error={false}
          helperText="This is a required field."
          label="Password"
          name="password"
          onChange={[MockFunction]}
          type="password"
          value="password"
        />
        <Memo(CommonButton)
          className="LoginModal-resetPassword-4"
          label=" Forgot your password?"
          onClick={[MockFunction]}
          variant="text"
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <Memo(CommonButton)
          disabled={false}
          label="Sign in"
          type="submit"
          variant="contained"
        />
        <Memo(CommonButton)
          className="LoginModal-signUpButton-6"
          label="Create an account"
          onClick={[MockFunction]}
          variant="text"
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </form>
</Memo(WithStyles(CommonDialog))>
`;
